client/errors.go

```diff
@@ -8,6 +8,11 @@ import (
 // ErrConnectionFailed is an error raised when the connection between the client and the server failed.
 var ErrConnectionFailed = errors.New("Cannot connect to the Docker daemon. Is the docker daemon running on this host?")
 
+// PermissionDenied returns an error with socketFile in the error message when a unix socket can not be opened.
+func PermissionDenied(socketFile string) error {
+	return fmt.Errorf("Cannot open %s. Permission Denied.", socketFile)
```

Comment:
I don't mind doing that at all, The reason I went with the PermissionDenied function like that is because it is following how the ErrorConnectionFailed function is setup. I don't mind making any changes you guys think are needed though.